scripts/build.js

```diff
@@ -0,0 +1,26 @@
+const {writeFile} = require('fs');
+const {resolve} = require('path');
+const {rollup} = require('rollup');
+const {promisify} = require('util');
+const config = require('../rollup.config');
+
+const writeFileAsync = promisify(writeFile);
+
+const cwd = process.cwd();
+const detection = "  if ('adoptedStyleSheets' in document) { return; }";
+
+(async () => {
+  try {
+    const bundle = await rollup(config);
+    const generated = await bundle.generate(config.output);
+    const {output: [{code}]} = generated;
+
+    const [first, second, ...other] = code.split('\n');
+    const result = [first, second, detection, ...other].join('\n');
+
+    await writeFileAsync(resolve(cwd, config.output.file), result, 'utf8');
+  } catch (e) {
+    console.error(e.stack);
+    process.exit(1);
+  }
+})();
```

Comment:
What is the purpose of this file?

Reply:
Unfortunately, buy the rollup itself I don't see any possibility to inject the check for implemented `adoptStyleSheets` in browser. I can do it in the index for, but also a lot it code like `WeakMap` creation will be excuted without need. So I used the direct injection of the code into the result of bundling. Now it detects implementation before any code runs.

Reply:
Assuming `index.js` gets bundled as part of the build, the 

```javascript
if ('adoptedStyleSheets' in document) {
  return;
}```

The build step should really just be `[npx]? rollup -c`.

Reply:
The problem is that index is the last file to bundle. So we will have something like that:
```js
(function() {
  'use strict'

  var adoptedSheetsRegistry = new WeakMap();
  var sheetMetadataRegistry = new WeakMap();
  var locationRegistry = new WeakMap();

  // A LOT of code, function delarations, etc.
  // Basically, there will be the whole polyfill code, because Rollup includes file by resolving 
  // their imports, where the entry point always comes at the end

  // Because we cannot `return` at the root of ES module
  if (!('adoptedStyleSheets' in document)) {
    // The small amount of content that is placed in `src/index.js`
    updatePrototype(OldCSSStyleSheet.prototype);

    window.CSSStyleSheet = ConstructStyleSheet;

    initAdoptedStyleSheets();

    if (document.readyState === 'loading') {
      document.addEventListener('DOMContentLoaded', initPolyfill);
    } else {
      initPolyfill();
    }
  }
})();
```
So, e.g., Chrome users will receive a polyfill that does a lot of work even if it is unnecessary. I would like to avoid this kind of situation. However, the only way I found is to inject the code after the build. I understand that it might look ugly, but it works.

Some other ways don't look like a solution as well.
- We can drop rollup and get back to the single file (however, it would harm maintainability).
- We can leave the detection to the user (which is against the common polyfill traditions).
- We can put detection code to every file (which would produce a lot of unnecessary code and if checks).

I chose the easiest and most efficient way. I consider this approach as using a rollup plugin. 

Hmm, probably, it would be better to put the detection code to the JS file in `src` instead of a string, it may improve maintainability by noting that the detection code does not appear from somewhere.

Reply:
Oh, I cannot put it to a file. It causes eslint to yell at me that `return` is used outside of a function. Well, then let it be a string.

Reply:
Got it. I'm not a huge fan of the approach, but can't think of anything better. This is looking really good, however with the caveat that `instanceof` doesn't work in IE11. 

I think that is acceptable for IE11 for the time being if we add that as a disclaimer in the README. I've glanced over the code and have no real issues, I'll start a deep dive now, but if you wouldn't mind adding that language to the README, I'll review and start the merge/release cycle.

Reply:
> however with the caveat that `instanceof` doesn't work in IE11

Yeah, but Babel has the [@babel/plugin-transform-instanceof](https://www.npmjs.com/package/@babel/plugin-transform-instanceof) to handle it. I used it for tests. We can mention that this plugin might be necessary in the docs.